chat_shell/chat_shell/services/chat_service.py

```
@@ -138,40 +128,29 @@ async def chat(self, request: ExecutionRequest) -> AsyncIterator[ExecutionEvent]
             # Process chat with the agent
             add_span_event("processing_chat_started")
-            logger.debug("[CHAT_SERVICE] Starting _process_chat...")
-            async for event in self._process_chat(request, core, state, emitter):
-                yield event
-
+            await self._process_chat(request, core, state, emitter)
             add_span_event("processing_chat_completed")
 
         except Exception as e:
             add_span_event("chat_error", {"error": str(e)})
             logger.exception("[CHAT_SERVICE] Exception in chat(): %s", e)
```

Ruff (0.15.0)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)